File: Module3/3A(JavaScript)/JSCodingChallenge/challenge4.js
// Print a table containing multiplication tables.

const table = 10;

// loop through rows
for (let a = 1; a <= table; a++) {
    let row = ''; // empty string for each row

    // loop through columns
    for (let b = 1; b <= table; b++) {
        // multiply the row by the column to get the value
        let value = a * b;

        // add the value to the row string with appropriate padding
        row += value.toString().padStart(4);
    }
    console.log(row);
}

File: Module4/Tasks_4C/PythonCodingChallenge/challenge4.py
# Function to print multiplication table
def multiplication_table():
    # Print the header row
    print("    ", end="")
    for i in range(1, 11):
        print(f"{i:4}", end="")
    print("\n" + " " * 4 + "-" * 40)

    # Print the table rows
    for i in range(1, 11):
        print(f"{i:2} |", end="")
        for j in range(1, 11):
            print(f"{i * j:4}", end="")
        print()

# Call the function to print the table
multiplication_table()
